### PR TITLE
chore: fix ci.yml canary releases to use nx, not lerna

### LINKED ISSUE #8960: Repo: CI failing to publish v8 canaries
### Suggestion

Following #8949, https://github.com/typescript-eslint/typescript-eslint/actions/runs/8758191820/job/24038987751:

```plaintext
Run npx lerna publish premajor --loglevel=verbose --canary --exact --force-publish --yes --dist-tag rc-v8
npm WARN exec The following package was not found and will be installed: lerna@8.1.2
lerna notice cli v8.1.2
lerna ERR! ENOLERNA `lerna.json` does not exist, have you run `lerna init`?
```

I've got the rename from lerna to nx in #8913 but hadn't applied it there whoops.

This is what I get for merging a CI change without consulting @JamesHenry! 😄 

💖

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8960
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

~I'm ... not sure which command we should use here 😬 is it ... `nx release`?~ ✅ 

Partially overlaps #8913.

💖

### CODE REVIEW

File: docs/maintenance/Releases.mdx [hunk 1]
```diff
@@ -26,15 +24,15 @@ Per [Users > Releases > Major Releases](../users/Releases.mdx#major-releases), w
    - Under `push:` > `branches:` at the beginning of the file, add an `- v${major}` list item.
    - Add a `publish_canary_version_v${major}` step the same as `publish_canary_version` except:
      - Change the `if` condition's branch check to: `if: github.ref == 'refs/heads/v${major}'`.
-     - Its publish command should be `npx lerna publish premajor --loglevel=verbose --canary --exact --force-publish --yes --dist-tag rc-v${major}`.
+     - Its publish command should be `npx nx publish premajor --loglevel=verbose --canary --exact --force-publish --yes --dist-tag rc-v${major}`.
```
Comment (JamesHenry) [inline docs/maintenance/Releases.mdx:27]: I forgot to update this when I updated the current implementation

**Review by JamesHenry — CHANGES_REQUESTED** [review]

File: docs/maintenance/Releases.mdx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply docs/maintenance/Releases.mdx:27]: @JamesHenry sorry, I'm not following - is there more you want to do?

Comment (JamesHenry) [reply docs/maintenance/Releases.mdx:27]: Yeah this doc needs to be updated to match the command I applied in the ci.yml, sorry I remembered when on mobile so noted this and I'm on mobile again :D

Comment (JoshuaKGoldberg) [reply docs/maintenance/Releases.mdx:27]: 😂 

I can update the doc and then merge, thanks!

**Pull request merged**